src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.SearchResult.cs

```diff
@@ -59,13 +60,32 @@ private class SearchResult : INavigateToSearchResult
                 });
             }
 
-            private static string ConstructSecondarySortString(DeclaredSymbolInfo declaredSymbolInfo)
+            private static readonly char[] s_dotArray = { '.' };
+
+            private static string ConstructSecondarySortString(
+                Document document,
+                DeclaredSymbolInfo declaredSymbolInfo)
             {
-                var secondarySortString = string.Concat(
-                    declaredSymbolInfo.ParameterCount.ToString("X4"),
-                    declaredSymbolInfo.TypeParameterCount.ToString("X4"),
-                    declaredSymbolInfo.Name);
-                return secondarySortString;
+                var parts = ArrayBuilder<string>.GetInstance();
+                try
+                {
+                    parts.Add(declaredSymbolInfo.ParameterCount.ToString("X4"));
+                    parts.Add(declaredSymbolInfo.TypeParameterCount.ToString("X4"));
+                    parts.Add(declaredSymbolInfo.Name);
+
+                    // For partial types, we break up the file name into pieces.  i.e. If we have
+                    // Outer.cs and Outer.Inner.cs  then we add "Outer" and "Outer Inner" to 
+                    // the secondary sort string.  That way "Outer.cs" will be weighted above
+                    // "Outer.Inner.cs"
```

Comment:
:memo: This [incorrectly](https://github.com/Microsoft/vs-threading/blob/v15.3/src/Microsoft.VisualStudio.Threading/JoinableTask+ExecutionQueue.cs) assumes that a `.` separates the containing and nested types in the file name, leading to a "works on my machine" scenario.